src/filelock/_api.py

```diff
@@ -143,6 +140,9 @@ def _initialize(  # noqa: PLR0913
             to pass the same object around.
 
         """
+        if hasattr(self, "_context"):
```

Comment:
I wonder if this will work with [this](https://github.com/pypa/virtualenv/blob/429d6a2319c1bf917452ca418bf041fe0daa130d/src/virtualenv/util/lock.py#L16)?

Reply:
Don't see why not, it's not setting the `_context` attribute is it? I also added a test for `is_singleton` along with some other tweaks in next push

Reply:
This will lead to a repeated call to `__init__` the inherited class whenever `lock_type(is_singleton=True)` is called, which is at least not obvious (also it will lead to the re-creation of internal objects in the heir)

Reply:
> I don't see how the superclass could force the subclass's __init__ to be bypassed?
It could be done with metaclass

Reply:
> Well, this's probably worth mentioning in the documentation if someone wants to make a singleton

Agree that this parameter should be better documented, it tripped me up for a minute as well. 

> It could be done with metaclass

Ah interesting, that could be cleaner. I've never used metaclasses. As I understand, you would define new in the metaclass to incorporate any overridden init behavior when a new object is actually created, and make the real init empty. 

Maybe worth looking into @gaborbernat? but above my pay grade